testsuite/test_wbemconnection_mock.py

```diff
@@ -4332,3 +4394,176 @@ def test_invokemethod(self, conn, ns, desc, inputs, exp_output,
+
+        assert schema.schema_version == DMTF_TEST_SCHEMA_VER
+        assert schema.schema_version_str == '2.49.0'
+        assert os.path.isdir(schema.schema_root_dir)
+        assert os.path.isdir(schema.schema_mof_dir)
+        assert os.path.isfile(schema.schema_mof_filename)
+        assert schema.schema_root_dir == test_schema_root_dir
+        mof_dir = 'mofFinal%s' % schema.schema_version_str
+        test_schema_mof_dir = os.path.join(test_schema_root_dir, mof_dir)
+        assert schema.schema_mof_dir == test_schema_mof_dir
+
+        assert schema.schema_mof_filename == \
+            os.path.join(test_schema_mof_dir,
+                         'cim_schema_%s.mof' % (schema.schema_version_str))
+
+        assert repr(schema) == 'DMTFCIMSchema(' \
+                               'schema_version=%s, ' \
+                               'schema_root_dir=%s, schema_mof_dir=%s, ' \
+                               'schema_mof_filename=%s)' % \
+                               (DMTF_TEST_SCHEMA_VER, test_schema_root_dir,
+                                schema.schema_mof_dir,
+                                schema.schema_mof_filename)
+
+        schema_mof = schema.build_schema_mof(['CIM_ComputerSystem', 'CIM_Door'])
+
+        exp_mof = '#pragma locale ("en_US")\n' \
+                  '#pragma include ("System/CIM_ComputerSystem.mof")\n' \
+                  '#pragma include ("Device/CIM_Door.mof")\n'
+
+        assert schema_mof == exp_mof
+
+    def test_schema_experimental_load(self, test_schema_root_dir):
+        # pylint: disable=no-self-use
+        """
+        Test the DMTFCIMSchema class and its methods to get a schema from the
+        DMTF, expand it, and to create a partial MOF schema definition for
+        compilation.
+        """
+        schema = DMTFCIMSchema(DMTF_TEST_SCHEMA_VER, test_schema_root_dir,
+                               use_experimental=True,
+                               verbose=True)
+
+        assert schema.schema_version == DMTF_TEST_SCHEMA_VER
+        assert schema.schema_version_str == '2.49.0'
+        assert os.path.isdir(schema.schema_root_dir)
+        assert os.path.isdir(schema.schema_mof_dir)
+        assert os.path.isfile(schema.schema_mof_filename)
+        assert schema.schema_root_dir == test_schema_root_dir
+        mof_dir = 'mofExperimental%s' % schema.schema_version_str
+        test_schema_mof_dir = os.path.join(test_schema_root_dir, mof_dir)
+        assert schema.schema_mof_dir == test_schema_mof_dir
+
+        assert schema.schema_mof_filename == \
+            os.path.join(test_schema_mof_dir,
+                         'cim_schema_%s.mof' % (schema.schema_version_str))
+
+    def test_two_schema_load(self, test_schema_root_dir):
+        # pylint: disable=no-self-use
+        """Test loading two different schema"""
+        schema1 = DMTFCIMSchema(DMTF_TEST_SCHEMA_VER, test_schema_root_dir,
+                                verbose=True)
```

Comment:
This test uses the schema version whose zip file we persist in the repo. So the test is that it creates all artefacts downstream from the zip file (e.g. the "mof*" directory, and the MOF files in it). While the assertions correctly test that these downstream artefacts exist afer the test was run (i.e. the ctor was invoked), the test preparation does not ensure that these downstream artefacts did **not** exist before the test was run.

I think we should ensure that as a test preparation. (i.e. delete all created artefacts, but not those that are persisted in the repo).

Reply:
Whoops.  As part of the changes to the test, created a test specifically to use the existing schema and confirm that it is still there when test done.

DONE